google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/StorageSnippets.java

```
@@ -1113,4 +1114,289 @@ public Bucket setDefaultKmsKey(String bucketName, String kmsKeyName) throws Stor
+    }
+    if (blob.getMetadata() != null) {
+      System.out.println("User metadata:");
+      for (Map.Entry<String, String> userMetadata : blob.getMetadata().entrySet()) {
+        System.out.println(userMetadata.getKey() + " " + userMetadata.getValue());
+      }
+    }
+    // [END storage_get_metadata]
+  }
+
+  /** Example of setting a retention policy on a bucket */
+  public Bucket setRetentionPolicy(String bucketName, Long retentionPeriod)
+      throws StorageException {
+    // [START storage_set_retention_policy]
+    // Instantiate a Google Cloud Storage client
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+
+    // The name of a bucket, e.g. "my-bucket"
+    // String bucketName = "my-bucket";
+
+    // The retetion period for objects in bucket
+    // 1 hour in seconds
+    // Long retentionPeriod = 3600L;
+
+    Bucket bucket =
+        storage.update(
+            BucketInfo.newBuilder(bucketName).setRetentionPeriod(retentionPeriod).build());
+
+    System.out.println(
+        "Retention period for " + bucketName + " is now " + bucket.getRetentionPeriod());
+    // [END storage_set_retention_policy]
+    return bucket;
+  }
+
+  /** Example of removing a retention policy on a bucket */
+  public Bucket removeRetentionPolicy(String bucketName) throws StorageException {
+    // [START storage_remove_retention_policy]
+    // Instantiate a Google Cloud Storage client
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+
+    // The name of a bucket, e.g. "my-bucket"
+    // String bucketName = "my-bucket";
+
+    Bucket bucket = storage.get(bucketName, BucketGetOption.fields(BucketField.RETENTION_POLICY));
+    if (bucket.retentionPolicyIsLocked() != null && bucket.retentionPolicyIsLocked()) {
+      System.out.println("Unable to remove retention period as retention policy is locked.");
+      return bucket;
```

A returned statement wont make sense here in the context of the snippet. Would it be better to throw an error instead?

---

Ack